07.ls_object/ls_file.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class LsFile
+  PERMISSION = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  attr_reader :file_name
+
+  def initialize(directory_path, file_name)
+    @directory_path = directory_path
+    @file_name = file_name
+    @stat = File.stat(file_path.to_s)
+  end
+
+  def type
+    @stat.ftype == 'file' ? '-' : @stat.ftype[0]
+  end
+
+  def permission
+    mode = @stat.mode.to_s(8)
+    (-3..-1)
+      .map do |i|
+        PERMISSION[mode[i]]
+      end
+      .join
```

Comment:
```suggestion
    (-3..-1).map { |i| PERMISSION[mode[i]] }.join
```

好みのレベルですが、これぐらいのコードなら1行で書いてもいいかも。

メソッドチェーンは改行した方が読みやすい、という場合でもmapは1行でいいかな〜。

```suggestion
    (-3..-1)
      .map { |i| PERMISSION[mode[i]] }
      .join
```